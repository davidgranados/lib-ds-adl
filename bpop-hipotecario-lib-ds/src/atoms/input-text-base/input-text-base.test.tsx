import React from "react";
import { mount, ReactWrapper } from "enzyme";
import InputTextBase from "./input-text-base";

describe("<InputTextBase />", () => {
  let wrapper: ReactWrapper<typeof InputTextBase>;
  const handleFocus = jest.fn();
  const handleBlur = jest.fn();
  const handleInput = jest.fn();
  beforeEach(() => {
    jest.clearAllMocks();
    // given
    wrapper = mount<typeof InputTextBase>(
      <InputTextBase id={"input-test-id"} onFocus={handleFocus} onBlur={handleBlur} onInput={handleInput} />
    );
  });
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should handle focus", () => {
    // when
    wrapper.find("#input-test-id").first().simulate("focus");
    // then
    expect(handleFocus).toBeCalled();
  });

  it("should handle blur", () => {
    // when
    wrapper.find("#input-test-id").first().simulate("blur");
    // then
    expect(handleBlur).toBeCalled();
  });

  it("should handle input", () => {
    // when
    wrapper.find("#input-test-id").first().simulate("input");
    // then
    expect(handleInput).toBeCalled();
  });
});
