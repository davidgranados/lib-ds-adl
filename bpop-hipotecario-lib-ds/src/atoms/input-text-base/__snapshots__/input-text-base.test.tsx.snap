// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputTextBase /> should match snapshot 1`] = `
<input
  className="default size-lg"
  id="input-test-id"
  onBlur={[Function]}
  onFocus={[MockFunction]}
  onInput={[Function]}
/>
`;
